sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusSessionReceiver.cs

```diff
@@ -100,6 +100,11 @@ public virtual async Task<byte[]> GetSessionStateAsync(CancellationToken cancell
             Argument.AssertNotClosed(IsDisposed, nameof(ServiceBusSessionReceiver));
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
             Logger.GetSessionStateStart(Identifier, SessionId);
+            using DiagnosticScope scope = _scopeFactory.CreateScope(
+                DiagnosticProperty.GetSessionStateActivityName,
+                sessionId: SessionId);
+            scope.Start();
```

Comment:
In case we get an exception, don't we wanna pass it in scope.Failed(exception) method in catch block?

Reply:
Also, we are not passing `sessionState` that this method is returning in one of the scope method, is this intentional?  I see we [pass the return value](https://github.com/Azure/azure-sdk-for-net/blob/d8de2f3d08d73c4ab3f606749789671b6b4b545d/sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs#L206) in `ServiceBusReceiver` operations.

Reply:
For sessionState, this is tricky since it is just bytes and we are no longer using the anonymous object args like in Track 1, so we need a string. We don't know the encoding so it isn't safe to log the string. We could consider having an overload that takes a string for this or use BinaryData.